File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        frq = {}
        q = deque()
        
        for i, c in enumerate(s):
            frq[c] = 1 + frq.get(c, 0)
            q.append((i, c))
        
        while q:
            i, c = q.popleft()
            if frq[c] == 1:
                return i
        
        return -1

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        HashMap<Character, Integer> freq = new HashMap<>();
        for(int i=0; i<s.length(); i++){
            if(freq.containsKey(s.charAt(i))){
                freq.put(s.charAt(i), freq.get(s.charAt(i))+1);
            }else{
                freq.put(s.charAt(i), 1);
            }
        }
        
        for(int i=0; i<s.length(); i++){
            if(freq.get(s.charAt(i)) == 1){
                return i;
            }
        }
        
        return -1;
    }
}